sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -390,21 +398,43 @@ def test_model_pipeline_same_as_normalize_true(LinearModel, params):
 # FIXME: 'normalize' to be removed in 1.2
 @pytest.mark.filterwarnings("ignore:'normalize' was deprecated")
 @pytest.mark.parametrize(
-    "estimator, is_sparse, with_mean",
-    [(LinearRegression, True, False),
-     (LinearRegression, False, True),
-     (LinearRegression, False, False)]
+    "estimator, params",
+    [
+         (Lasso, {"tol": 1e-16, "alpha": 0.1}),
+         (RidgeClassifier, {"solver": 'sparse_cg', "alpha": 0.1}),
+         (ElasticNet, {"tol": 1e-16, 'l1_ratio': 1, "alpha": 0.1}),
+         (ElasticNet, {"tol": 1e-16, 'l1_ratio': 0, "alpha": 0.1}),
+         (Ridge, {"solver": 'sparse_cg', 'tol': 1e-12, "alpha": 0.1}),
+         (LinearRegression, {}),
+     ]
+)
+@pytest.mark.parametrize(
+    "is_sparse, with_mean", [
+        (False, True),
+        (False, False),
+        (True, False)
+        # No need to test sparse and with_mean=True
+    ]
 )
 def test_linear_model_sample_weights_normalize_in_pipeline(
-        estimator, is_sparse, with_mean
+        is_sparse, with_mean, estimator, params
 ):
-    # Test that the results for running linear regression LinearRegression with
-    # sample_weight set and with normalize set to True gives similar results as
-    # LinearRegression with no normalize in a pipeline with a StandardScaler
-    # and set sample_weight.
+    # Test that the results for running linear model with sample_weight
+    # and with normalize set to True gives similar results as the same linear
+    # model with normalize set to False in a pipeline with
+    # a StandardScaler and sample_weight.
+    model_name = estimator.__name__
+
+    if model_name in ['Lasso', 'ElasticNet'] and is_sparse:
+        pytest.skip(f'{model_name} does not support sample_weight with sparse')
+
     rng = np.random.RandomState(0)
     X, y = make_regression(n_samples=20, n_features=5, noise=1e-2,
                            random_state=rng)
+
+    if is_classifier(estimator):
+        y = np.sign(y)
+
     # make sure the data is not centered to make the problem more
     # difficult
     X += 10
```

Comment:
Instead of this we could do something like:

```python
X[X<0] = 0
```

this would both shift the mean and make the sparse data case actually use sparse data.